src/Util/HTTPHeaders.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * @file src/Util/HTTPHeaders.php
+ */
+namespace Friendica\Util;
+
+/**
+ * Ported from Hubzilla: https://framagit.org/hubzilla/core/blob/master/Zotlabs/Web/HTTPHeaders.php
+ */
+class HTTPHeaders
+{
+	private $in_progress = [];
+	private $parsed = [];
+
+	function __construct($headers)
+	{
+		$lines = explode("\n", str_replace("\r", '', $headers));
+
+		if ($lines) {
+			foreach ($lines as $line) {
+				if (preg_match('/^\s+/', $line, $matches) && trim($line)) {
+					if ($this->in_progress['k']) {
```

Comment:
This will trigger a PHP Notice "index 'k' not found". Please use `isset()` or `!empty()`.